### PR TITLE
Add consumer onboarding overview page behind feature flag

### PR DESCRIPTION
**Pull request by mjlumetta**
### Description

adds the overview page to the consumer onboarding docs, plus the structure like main nav changes and the consumer onboarding side nav. completes [API-7687](https://vajira.max.gov/browse/API-7687).

key links:

- [Figma design](https://www.figma.com/file/9sOD2GDAsyNTayvOqKlcJV/API-3854-Consumer?node-id=1397%3A0)
- [content](https://docs.google.com/document/d/1zjJIHPUrwug8fAGc1QEYGmb_8HCAMZ5Ymcq6UwN06vs/edit#)

### Process

- [x] Tests added or updated for change
- [ ] Docs updated
  - [ ] If you have made structural changes to the site, the [Developer Portal Content Types](https://community.max.gov/x/qUwOg) docs have been updated
  - [n/a] If you have made/confirmed any decisions about user interactions and accessibility, the [Accessible Component Design](https://community.max.gov/x/xS8mgg) docs have been updated
- [x] Accessibility concerns have been considered and addressed
  
  - [x] If you've made any UI changes, you've run an axe check using the axe browser extension that reported no new issues

### Requested feedback

wondering if anyone has opinions on whether or not to use markdown for some part of this functionality. not sure where we are with that at the moment.

### CODE REVIEW

**Comment by brentschneider** [comment]
The desktop and mobile layouts look fantastic from my perspective @mjlumetta.

Regarding using markdown? honestly, up to your discretion what's most convenient to implement and maintain. The Overview and Prepare for the demo pages are the least complicated regarding mixing components, whereas the other pages do get a little squirrely.

**Review by darduda — DISMISSED** [review]
Looks great

File: src/containers/consumerOnboarding/OnboardingOverview.tsx [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+          <a href={CONSUMER_SANDBOX_PATH}>request an API key</a>.
+        </p>
+      </li>
+      <li className="process-step list-two" aria-labelledby="request-prod-access">
+        <strong id="request-prod-access">Request production access</strong>
+        <p>
+          Getting production access can take less than a week to more than a month, depending on the API.
+        </p>
+        <p>
+          <a href={CONSUMER_PROD_PATH}>Learn what’s needed on the production access form.</a>
```
Comment (ndsprinkle) [inline src/containers/consumerOnboarding/OnboardingOverview.tsx:34]: Small detail, but is the link supposed to be in a separate paragraph or in the same? On Figma, it looks like it's supposed to be inline like the other steps. 

Although I am separately confused by Figma as to why the step 4 link is only half of the sentence ("Learn more about" is not a link and "working with our APIs" is) while this step 2 link is the whole sentence ("Learn what’s needed on the production access form.").

File: src/containers/consumerOnboarding/ConsumerOnboardingRoot.test.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    );
+  });
+
+  describe('side nav', () => {
+    it('renders the side nav', () => {
+      const nav = screen.getByRole('navigation', { name: 'Consumer Onboarding Page Nav' });
+      expect(nav).toBeInTheDocument();
+    });
+
+    it.each([
```
Comment (ndsprinkle) [inline src/containers/consumerOnboarding/ConsumerOnboardingRoot.test.tsx:29]: I feel dumb for not knowing this existed, but `it.each` (and `describe.each`) is awesome. Going to have to remember this to use in the future.

File: src/components/sideNav/SideNav.tsx [hunk 3]
```diff
@@ -69,6 +69,7 @@ const SideNav = (props: SideNavProps): JSX.Element => {
             'vads-u-border-bottom--2px',
             'vads-u-border-left--2px',
             'vads-u-border-right--2px',
+            'vads-u-border-top--0',
```
Comment (ndsprinkle) [inline src/components/sideNav/SideNav.tsx:72]: I had noticed this thicker border on top of the side navs before, but assumed it was intentional. Apparently not :)

File: src/containers/consumerOnboarding/OnboardingOverview.tsx (hunk 1 above)
Comment (ndsprinkle) [reply src/containers/consumerOnboarding/OnboardingOverview.tsx:34]: Ah, now I see that in the content doc it does live in a separate paragraph. https://docs.google.com/document/d/1zjJIHPUrwug8fAGc1QEYGmb_8HCAMZ5Ymcq6UwN06vs/edit

I don't know then, not sure which one is the source of truth.

**Review by ndsprinkle — DISMISSED** [review]

**Review by ndsprinkle — APPROVED** [review]

**Pull request merged**